From: Jan Braun <janbraun@gmx.net>
Subject: [PATCH] fixes/cli-output-verbosity

Common unix convention for CLI programs is being silent for successful
operations, and being verbose on errors. This patch adjust the output verbosity
to match the unix conventions.

Signed-off-by: Benjamin Drung <bdrung@ubuntu.com>

---
 src/clients/cli/cmd_pls.c |   10 ++++++----
 src/clients/cli/common.h  |    2 ++
 src/clients/cli/main.c    |   10 ++++++++++
 3 files changed, 18 insertions(+), 4 deletions(-)

diff --git a/src/clients/cli/cmd_pls.c b/src/clients/cli/cmd_pls.c
index 967dea1..bd07efe 100644
--- a/src/clients/cli/cmd_pls.c
+++ b/src/clients/cli/cmd_pls.c
@@ -37,7 +37,7 @@ add_item_to_playlist (xmmsc_connection_t *conn, gchar *playlist, gchar *item)
 
 	url = format_url (item, G_FILE_TEST_IS_REGULAR);
 	if (!url) {
-		print_error ("Invalid url");
+		print_error ("Invalid url: %s", item);
 	}
 
 	res = xmmsc_playlist_add_url (conn, playlist, url);
@@ -50,7 +50,9 @@ add_item_to_playlist (xmmsc_connection_t *conn, gchar *playlist, gchar *item)
 	}
 	xmmsc_result_unref (res);
 
-	print_info ("Added %s", item);
+	if (verbose) {
+        print_info ("Added %s", item);
+    }
 }
 
 
@@ -250,7 +252,7 @@ cmd_addarg (xmmsc_connection_t *conn, gint argc, gchar **argv)
 	if (!url) {
 		url = format_url (argv[3], G_FILE_TEST_IS_REGULAR);
 		if (!url) {
-			print_error ("Invalid url");
+			print_error ("Invalid url: %s", argv[3]);
 		} else {
 			/* FIXME: Fulhack to check for optional playlist argument */
 			playlist = argv[2];
@@ -296,7 +298,7 @@ cmd_insert (xmmsc_connection_t *conn, gint argc, gchar **argv)
 	}
 
 	if (!url) {
-		print_error ("Invalid url");
+		print_error ("Invalid url: %s", (*endptr == '\0') ? argv[3] : argv[4] );
 	}
 
 	res = xmmsc_playlist_insert_url (conn, playlist, pos, url);
diff --git a/src/clients/cli/common.h b/src/clients/cli/common.h
index a1ce220..0fd46bb 100644
--- a/src/clients/cli/common.h
+++ b/src/clients/cli/common.h
@@ -31,6 +31,8 @@ typedef struct {
 	void (*func) (xmmsc_connection_t *conn, int argc, char **argv);
 } cmds;
 
+gboolean verbose;
+
 gboolean x_realpath (const gchar *item, gchar *rpath);
 gchar *x_path2url (gchar *path);
 gchar *format_url (gchar *item, GFileTest test);
diff --git a/src/clients/cli/main.c b/src/clients/cli/main.c
index 65cf7ea..5b239f3 100644
--- a/src/clients/cli/main.c
+++ b/src/clients/cli/main.c
@@ -238,6 +238,16 @@ main (gint argc, gchar **argv)
 
 	setlocale (LC_ALL, "");
 
+    verbose = FALSE;
+    while ( argc >1 && ( !g_ascii_strcasecmp (argv[1], "-v") ||
+                    !g_ascii_strcasecmp (argv[1], "--verbose") )
+    ) {
+        verbose = TRUE;
+        argv[1] = argv[0];
+        --argc;
+        ++argv;
+    }
+
 	config = read_config ();
 	atexit (free_config);
 
-- 
tg: (a2ecaf9..) fixes/cli-output-verbosity (depends on: upstream)
